src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -229,6 +241,29 @@ public static void removeProperty(final PublicJSONPointer property, final JSONOb
         }
     }
 
+    public static boolean compareProperty(final PublicJSONPointer property,
+                                          final JSONObject previousResourceModel,
+                                          final JSONObject currentresourceModel) {
+        List<String> refTokens = property.getRefTokens();
+        final String key = refTokens.get(refTokens.size() - 1);
+        try {
+            // if size is more than one, fetch parent object/array of key to remove so that
+            // we can remove
+            if (refTokens.size() > 1) {
+                // use sublist to specify to point at the parent object
+                final JSONPointer parentObjectPointer = new JSONPointer(refTokens.subList(0, refTokens.size() - 1));
+                final JSONObject previousParentObject = (JSONObject) parentObjectPointer.queryFrom(previousResourceModel);
+                final JSONObject currentParentObject = (JSONObject) parentObjectPointer.queryFrom(currentresourceModel);
+                return previousParentObject.get(key) == currentParentObject.get(key);
+            } else {
+                return previousResourceModel.get(key) == currentresourceModel.get(key);
+            }
+        } catch (JSONPointerException | NumberFormatException e) {
+            // do nothing, as this indicates the model does not have a value for the pointer
```

Comment:
if both don't have a property would it trip this case?

Reply:
thats a good catch, we should not be failing in this condition